src/test/java/com/fittracker/fittracker/controller/ProductControllerValidationTest.java

```diff
@@ -0,0 +1,174 @@
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import com.fittracker.fittracker.exception.ErrorResponseMapper;
+import com.fittracker.fittracker.request.ProductRequest;
+import com.fittracker.fittracker.security.JwtAuthenticationFilter;
+import com.fittracker.fittracker.service.ProductService;
+import java.net.URI;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.verification.VerificationMode;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Import;
+import org.springframework.http.HttpStatus;
+import org.springframework.test.web.servlet.MockMvc;
+import org.testcontainers.shaded.com.fasterxml.jackson.databind.ObjectMapper;
+
+@AutoConfigureMockMvc(addFilters = false)
+@Import(ErrorResponseMapper.class)
+@WebMvcTest(controllers = ProductController.class, excludeFilters = {
+    @ComponentScan.Filter(type = ASSIGNABLE_TYPE, value = JwtAuthenticationFilter.class)})
+public class ProductControllerValidationTest {
+
+    @MockBean
+    private ProductService productService;
+    @Autowired
+    private MockMvc mockMvc;
+
+    private static final String ENDPOINT = "/api/v1/product";
+
+    private static final ObjectMapper mapper = new ObjectMapper();
+
+    @BeforeEach
+    void beforeEach() {
+        mapper.setVisibility(FIELD, ANY);
+    }
+
+    @Nested
+    class Get {
+
+        @ParameterizedTest
+        @MethodSource("getProductRequestTestDataProvider")
+        void givenGetUriRequest_shouldReturnResponse(String params, HttpStatus responseStatus, String responseBody,
+            VerificationMode numberOfServiceInvocations) throws Exception {
+            mockMvc
+                .perform(get(URI.create(ENDPOINT + params)))
+                .andExpect(status().is(responseStatus.value()))
+                .andExpect(content().string(responseBody));
+
+            verify(productService, numberOfServiceInvocations).findById(any());
+        }
+
+        private static Stream<Arguments> getProductRequestTestDataProvider() {
+            return Stream.of(
+                of("/notUUID", BAD_REQUEST, "{\"field\":\"id\",\"message\":\"Invalid data type\"}", never()),
+                of("/57a407b2-4a54-4ae7-b93c-9e3227af26bg", BAD_REQUEST,
```

Comment:
why is it invalid UUID?

Reply:
I accidentally discovered that last letter of alphabet that can appear in last section of any UUID is "f". I changed last to be "g" to check if validation is not only checking lengths of sections but also well defined set of characters.

https://www.uuidtools.com/what-is-uuid ( UUID Regular Expression (REGEX) ) at the bottom of the page.